src/black/__init__.py

```diff
@@ -2083,8 +2083,16 @@ def visit_STRING(self, leaf: Leaf) -> Iterator[Line]:
         ) and is_multiline_string(leaf):
             prefix = "    " * self.current_line.depth
             docstring = fix_docstring(leaf.value[3:-3], prefix)
+            quote_char = leaf.value[0]
+            if len(docstring) > 0 and docstring[-1] == quote_char:
```

Comment:
```suggestion
            if docstring and docstring[-1] == quote_char:
```

since the only string evaluating to false is `""`

Reply:
If we go that way in this change, sure.  @JelleZijlstra seemed to prefer a solution which preserved leading/trailing whitespace.  I've moved that code, previously only in a comment above, into the PR.